The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12329051264/job/34412974891) [(plain)](https://github.com/rust-lang/rust/commit/a5188c25c9fe310f5ce75aa3dd3ef95fdacf47f2/checks/34412974891/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#16 exporting to docker image format
#16 sending tarball 26.9s done
#16 DONE 32.5s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
---- [ui] tests/ui-fulldeps/run-compiler-twice.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/run-compiler-twice.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/run-compiler-twice/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
error[E0599]: no method named `enter` found for reference `&Compiler` in the current scope
##[error]  --> /checkout/tests/ui-fulldeps/run-compiler-twice.rs:80:31
   |
   |
LL |         let linker = compiler.enter(|queries| {
   |                      ---------^^^^^ method not found in `&Compiler`
error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0599`.
------------------------------------------
------------------------------------------


---- [ui] tests/ui-fulldeps/stable-mir/check_abi.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_abi.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_abi/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#103), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_abi.rs:157:5
   |
   |
LL |     run!(args, test_stable_mir).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_attribute.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_attribute.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_attribute/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#25), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_attribute.rs:109:5
   |
   |
LL |     run!(args, test_stable_mir).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_binop.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_binop.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_binop/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#121), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_binop.rs:87:5
   |
   |
LL |     run!(args, test_binops).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_def_ty.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_def_ty.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_def_ty/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#146), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_def_ty.rs:88:5
   |
   |
LL |     run!(args, test_def_tys).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_allocation.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_allocation.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_allocation/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#190), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_allocation.rs:231:5
   |
   |
LL |     run!(args, test_stable_mir).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_crate_defs.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_crate_defs.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_crate_defs/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#21), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_crate_defs.rs:96:5
   |
   |
LL |     run!(args, test_stable_mir).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_foreign.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_foreign.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_foreign/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#133), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_foreign.rs:71:5
   |
   |
LL |     run!(args, || test_foreign()).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_defs.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_defs.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_defs/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#150), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_defs.rs:124:5
   |
   |
LL |     run!(args, test_stable_mir).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_instance.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_instance.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_instance/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#151), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_instance.rs:100:5
   |
   |
LL |     run!(args, test_stable_mir).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_item_kind.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_item_kind.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_item_kind/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#26), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_item_kind.rs:60:5
   |
   |
LL |     run!(args, test_item_kind).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_intrinsics.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_intrinsics.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_intrinsics/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#152), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_intrinsics.rs:123:5
   |
   |
LL |     run!(args, test_intrinsics).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_normalization.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_normalization.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_normalization/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#130), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_normalization.rs:74:5
   |
   |
LL |     run!(args, test_stable_mir).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_trait_queries.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_trait_queries.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_trait_queries/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#33), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_trait_queries.rs:84:5
   |
   |
LL |     run!(args, test_traits).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_ty_fold.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_ty_fold.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_ty_fold/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#121), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_ty_fold.rs:88:5
   |
   |
LL |     run!(args, test_stable_mir).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/compilation-result.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/compilation-result.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/compilation-result/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#7), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/compilation-result.rs:39:18
   |
   |
LL |     let result = run!(args, || ControlFlow::Continue::<(), bool>(true));
   |                  |
   |                  no `Queries` in the root
   |                  help: a similar name exists in the module: `queries`
   |
   |
   = note: this error originates in the macro `run_driver` which comes from the expansion of the macro `run` (in Nightly builds, run with -Z macro-backtrace for more info)

 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#10), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/compilation-result.rs:44:18
   |
   |
LL |     let result = run!(args, || ControlFlow::Break::<bool, i32>(false));
   |                  |
   |                  no `Queries` in the root
   |                  help: a similar name exists in the module: `queries`
   |
   |
   = note: this error originates in the macro `run_driver` which comes from the expansion of the macro `run` (in Nightly builds, run with -Z macro-backtrace for more info)

 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#13), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/compilation-result.rs:51:18
   |
   |
LL |     let result = run!(args, || unreachable!() as ControlFlow<()>);
   |                  |
   |                  no `Queries` in the root
   |                  help: a similar name exists in the module: `queries`
   |
   |
   = note: this error originates in the macro `run_driver` which comes from the expansion of the macro `run` (in Nightly builds, run with -Z macro-backtrace for more info)

 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#19), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/compilation-result.rs:58:18
   |
   |
LL |     let result = run!(args, || unreachable!() as ControlFlow<()>);
   |                  |
   |                  no `Queries` in the root
   |                  help: a similar name exists in the module: `queries`
   |
   |
   = note: this error originates in the macro `run_driver` which comes from the expansion of the macro `run` (in Nightly builds, run with -Z macro-backtrace for more info)

 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#25), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/compilation-result.rs:65:18
   |
   |
LL |     let result = run!(args, || ControlFlow::Continue::<(), usize>(captured.len()));
   |                  |
   |                  no `Queries` in the root
   |                  help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/check_transform.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/check_transform.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/check_transform/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#133), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/check_transform.rs:132:5
   |
   |
LL |     run!(args, test_transform).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/projections.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/projections.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/projections/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#152), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/projections.rs:160:5
   |
   |
LL |     run!(args, test_place_projections).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/smir_internal.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/smir_internal.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/smir_internal/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#95), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/smir_internal.rs:50:5
   |
   |
LL |     run_with_tcx!(args, test_translation).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
   |
   = note: this error originates in the macro `run_driver` which comes from the expansion of the macro `run_with_tcx` (in Nightly builds, run with -Z macro-backtrace for more info)
error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0432`.
------------------------------------------
------------------------------------------


---- [ui] tests/ui-fulldeps/stable-mir/crate-info.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/crate-info.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/crate-info/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#215), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/crate-info.rs:195:5
   |
   |
LL |     run!(args, test_stable_mir).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
---
---- [ui] tests/ui-fulldeps/stable-mir/smir_serde.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/smir_serde.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/smir_serde/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#101), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/smir_serde.rs:61:5
   |
   |
LL |     run_with_tcx!(args, serialize_to_json).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
   |
   = note: this error originates in the macro `run_driver` which comes from the expansion of the macro `run_with_tcx` (in Nightly builds, run with -Z macro-backtrace for more info)
error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0432`.
------------------------------------------
------------------------------------------


---- [ui] tests/ui-fulldeps/stable-mir/smir_visitor.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui-fulldeps/stable-mir/smir_visitor.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--check-cfg" "cfg(FALSE)" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui-fulldeps/stable-mir/smir_visitor/a" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--edition=2021"
--- stderr -------------------------------
--- stderr -------------------------------
 WARN rustc_errors::emitter Invalid span /rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs:317:42: 317:49 (#46), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/FAKE_PREFIX/compiler/rustc_smir/src/rustc_internal/mod.rs" }) }
##[error]  --> /checkout/tests/ui-fulldeps/stable-mir/smir_visitor.rs:117:5
   |
   |
LL |     run!(args, test_visitor).unwrap();
   |     |
   |     no `Queries` in the root
   |     help: a similar name exists in the module: `queries`
   |
```

</details>

--- a/compiler/rustc_driver_impl/src/lib.rs
+++ b/compiler/rustc_driver_impl/src/lib.rs
@@ -394,7 +394,7 @@ fn run_compiler(
         // If pretty printing is requested: Figure out the representation, print it and exit
         if let Some(pp_mode) = sess.opts.pretty {
             if pp_mode.needs_ast_map() {
-                create_and_enter_global_ctxt(&compiler, krate, |tcx| {
+                create_and_enter_global_ctxt(compiler, krate, |tcx| {
                     tcx.ensure().early_lint_checks(());
                     pretty::print(sess, pp_mode, pretty::PrintExtra::NeedsAstMap { tcx });
                     passes::write_dep_info(tcx);
@@ -414,7 +414,7 @@ fn run_compiler(
             return early_exit();
         }
 
-        let linker = create_and_enter_global_ctxt(&compiler, krate, |tcx| {
+        let linker = create_and_enter_global_ctxt(compiler, krate, |tcx| {
             let early_exit = || {
                 sess.dcx().abort_if_errors();
                 None

--- a/compiler/rustc_driver_impl/src/lib.rs
+++ b/compiler/rustc_driver_impl/src/lib.rs
@@ -45,7 +45,7 @@ use rustc_errors::registry::Registry;
 use rustc_errors::{ColorConfig, DiagCtxt, ErrCode, FatalError, PResult, markdown};
 use rustc_feature::find_gated_cfg;
 use rustc_interface::util::{self, get_codegen_backend};
-use rustc_interface::{Linker, interface, passes};
+use rustc_interface::{Linker, create_and_enter_global_ctxt, interface, passes};
 use rustc_lint::unerased_lint_store;
 use rustc_metadata::creader::MetadataLoader;
 use rustc_metadata::locator;
@@ -400,7 +400,9 @@ fn run_compiler(
             // If pretty printing is requested: Figure out the representation, print it and exit
             if let Some(pp_mode) = sess.opts.pretty {
                 if pp_mode.needs_ast_map() {
-                    queries.global_ctxt().enter(|tcx| {
+                    let krate = queries.parse().steal();
+
+                    create_and_enter_global_ctxt(&compiler, krate, |tcx| {
                         tcx.ensure().early_lint_checks(());
                         pretty::print(sess, pp_mode, pretty::PrintExtra::NeedsAstMap { tcx });
                         passes::write_dep_info(tcx);
@@ -425,7 +427,9 @@ fn run_compiler(
                 return early_exit();
             }
 
-            queries.global_ctxt().enter(|tcx| {
+            let krate = queries.parse().steal();
+
+            create_and_enter_global_ctxt(&compiler, krate, |tcx| {
                 // Make sure name resolution and macro expansion is run.
                 let _ = tcx.resolver_for_lowering();
 

--- a/compiler/rustc_interface/src/lib.rs
+++ b/compiler/rustc_interface/src/lib.rs
@@ -8,7 +8,7 @@
 // tidy-alphabetical-end
 
 mod callbacks;
-mod errors;
+pub mod errors;
 pub mod interface;
 pub mod passes;
 mod proc_macro_decls;
@@ -17,7 +17,7 @@ pub mod util;
 
 pub use callbacks::setup_callbacks;
 pub use interface::{Config, run_compiler};
-pub use passes::DEFAULT_QUERY_PROVIDERS;
+pub use passes::{DEFAULT_QUERY_PROVIDERS, create_and_enter_global_ctxt};
 pub use queries::{Linker, Queries};
 
 #[cfg(test)]

--- a/compiler/rustc_interface/src/passes.rs
+++ b/compiler/rustc_interface/src/passes.rs
@@ -709,25 +709,10 @@ pub static DEFAULT_QUERY_PROVIDERS: LazyLock<Providers> = LazyLock::new(|| {
     *providers
 });
 
-pub fn create_and_enter_global_ctxt<T>(
+pub fn create_and_enter_global_ctxt<T, F: for<'tcx> FnOnce(TyCtxt<'tcx>) -> T>(
     compiler: &Compiler,
-    krate: rustc_ast::Crate,
-    f: impl for<'tcx> FnOnce(TyCtxt<'tcx>) -> T,
-) -> T {
-    let gcx_cell = OnceLock::new();
-    let arena = WorkerLocal::new(|_| Arena::default());
-    let hir_arena = WorkerLocal::new(|_| rustc_hir::Arena::default());
-
-    create_and_enter_global_ctxt_inner(compiler, krate, &gcx_cell, &arena, &hir_arena, f)
-}
-
-fn create_and_enter_global_ctxt_inner<'tcx, T>(
-    compiler: &'tcx Compiler,
     mut krate: rustc_ast::Crate,
-    gcx_cell: &'tcx OnceLock<GlobalCtxt<'tcx>>,
-    arena: &'tcx WorkerLocal<Arena<'tcx>>,
-    hir_arena: &'tcx WorkerLocal<rustc_hir::Arena<'tcx>>,
-    f: impl FnOnce(TyCtxt<'tcx>) -> T,
+    f: F,
 ) -> T {
     let sess = &compiler.sess;
 
@@ -776,8 +761,25 @@ fn create_and_enter_global_ctxt_inner<'tcx, T>(
 
     let incremental = dep_graph.is_fully_enabled();
 
-    let qcx = gcx_cell.get_or_init(move || {
+    let gcx_cell = OnceLock::new();
+    let arena = WorkerLocal::new(|_| Arena::default());
+    let hir_arena = WorkerLocal::new(|_| rustc_hir::Arena::default());
+
+    // This closure is necessary to force rustc to perform the correct lifetime
+    // subtyping for GlobalCtxt::enter to be allowed.
+    let inner: Box<
+        dyn for<'tcx> FnOnce(
+            &'tcx Compiler,
+            &'tcx OnceLock<GlobalCtxt<'tcx>>,
+            &'tcx WorkerLocal<Arena<'tcx>>,
+            &'tcx WorkerLocal<rustc_hir::Arena<'tcx>>,
+            F,
+        ) -> T,
+    > = Box::new(move |compiler, gcx_cell, arena, hir_arena, f| {
+        let sess = &compiler.sess;
+
         TyCtxt::create_global_ctxt(
+            gcx_cell,
             sess,
             crate_types,
             stable_crate_id,
@@ -794,28 +796,29 @@ fn create_and_enter_global_ctxt_inner<'tcx, T>(
             ),
             providers.hooks,
             compiler.current_gcx.clone(),
+            |tcx| {
+                let feed = tcx.create_crate_num(stable_crate_id).unwrap();
+                assert_eq!(feed.key(), LOCAL_CRATE);
+                feed.crate_name(crate_name);
+
+                let feed = tcx.feed_unit_query();
+                feed.features_query(tcx.arena.alloc(rustc_expand::config::features(
+                    sess,
+                    &pre_configured_attrs,
+                    crate_name,
+                )));
+                feed.crate_for_resolver(tcx.arena.alloc(Steal::new((krate, pre_configured_attrs))));
+                feed.output_filenames(Arc::new(outputs));
+
+                let res = f(tcx);
+                // FIXME maybe run finish even when a fatal error occured? or at least tcx.alloc_self_profile_query_strings()?
+                tcx.finish();
+                res
+            },
         )
     });
 
-    qcx.enter(|tcx| {
-        let feed = tcx.create_crate_num(stable_crate_id).unwrap();
-        assert_eq!(feed.key(), LOCAL_CRATE);
-        feed.crate_name(crate_name);
-
-        let feed = tcx.feed_unit_query();
-        feed.features_query(tcx.arena.alloc(rustc_expand::config::features(
-            sess,
-            &pre_configured_attrs,
-            crate_name,
-        )));
-        feed.crate_for_resolver(tcx.arena.alloc(Steal::new((krate, pre_configured_attrs))));
-        feed.output_filenames(Arc::new(outputs));
-
-        let res = f(tcx);
-        // FIXME maybe run finish even when a fatal error occured? or at least tcx.alloc_self_profile_query_strings()?
-        tcx.finish();
-        res
-    })
+    inner(compiler, &gcx_cell, &arena, &hir_arena, f)
 }
 
 /// Runs all analyses that we guarantee to run, even if errors were reported in earlier analyses.

--- a/compiler/rustc_interface/src/passes.rs
+++ b/compiler/rustc_interface/src/passes.rs
@@ -41,7 +41,7 @@ use tracing::{info, instrument};
 use crate::interface::Compiler;
 use crate::{errors, proc_macro_decls, util};
 
-pub(crate) fn parse<'a>(sess: &'a Session) -> ast::Crate {
+pub fn parse<'a>(sess: &'a Session) -> ast::Crate {
     let krate = sess
         .time("parse_crate", || {
             let mut parser = unwrap_or_emit_fatal(match &sess.io.input {
@@ -709,7 +709,22 @@ pub static DEFAULT_QUERY_PROVIDERS: LazyLock<Providers> = LazyLock::new(|| {
     *providers
 });
 
-pub(crate) fn create_global_ctxt<'tcx>(
+pub fn create_and_enter_global_ctxt<'tcx, T>(
+    compiler: &'tcx Compiler,
+    krate: rustc_ast::Crate,
+    f: impl for<'a> FnOnce(TyCtxt<'a>) -> T,
+) -> T {
+    let gcx_cell = OnceLock::new();
+    let arena = WorkerLocal::new(|_| Arena::default());
+    let hir_arena = WorkerLocal::new(|_| rustc_hir::Arena::default());
+
+    let gcx = create_global_ctxt(compiler, krate, &gcx_cell, &arena, &hir_arena);
+    let ret = gcx.enter(f);
+    gcx.finish();
+    ret
+}
+
+fn create_global_ctxt<'tcx>(
     compiler: &'tcx Compiler,
     mut krate: rustc_ast::Crate,
     gcx_cell: &'tcx OnceLock<GlobalCtxt<'tcx>>,

--- a/compiler/rustc_interface/src/queries.rs
+++ b/compiler/rustc_interface/src/queries.rs
@@ -7,9 +7,7 @@ use rustc_codegen_ssa::CodegenResults;
 use rustc_codegen_ssa::traits::CodegenBackend;
 use rustc_data_structures::steal::Steal;
 use rustc_data_structures::svh::Svh;
-use rustc_data_structures::sync::{OnceLock, WorkerLocal};
 use rustc_hir::def_id::LOCAL_CRATE;
-use rustc_middle::arena::Arena;
 use rustc_middle::dep_graph::DepGraph;
 use rustc_middle::ty::{GlobalCtxt, TyCtxt};
 use rustc_session::Session;
@@ -65,50 +63,17 @@ impl<'a, 'tcx> QueryResult<'a, &'tcx GlobalCtxt<'tcx>> {
 
 pub struct Queries<'tcx> {
     compiler: &'tcx Compiler,
-    gcx_cell: OnceLock<GlobalCtxt<'tcx>>,
-
-    arena: WorkerLocal<Arena<'tcx>>,
-    hir_arena: WorkerLocal<rustc_hir::Arena<'tcx>>,
 
     parse: Query<ast::Crate>,
-    // This just points to what's in `gcx_cell`.
-    gcx: Query<&'tcx GlobalCtxt<'tcx>>,
 }
 
 impl<'tcx> Queries<'tcx> {
     pub fn new(compiler: &'tcx Compiler) -> Queries<'tcx> {
-        Queries {
-            compiler,
-            gcx_cell: OnceLock::new(),
-            arena: WorkerLocal::new(|_| Arena::default()),
-            hir_arena: WorkerLocal::new(|_| rustc_hir::Arena::default()),
-            parse: Query { result: RefCell::new(None) },
-            gcx: Query { result: RefCell::new(None) },
-        }
-    }
-
-    pub fn finish(&'tcx self) {
-        if let Some(gcx) = self.gcx_cell.get() {
-            gcx.finish();
-        }
+        Queries { compiler, parse: Query { result: RefCell::new(None) } }
     }
 
     pub fn parse(&self) -> QueryResult<'_, ast::Crate> {
         self.parse.compute(|| passes::parse(&self.compiler.sess))
-    }
-
-    pub fn global_ctxt(&'tcx self) -> QueryResult<'tcx, &'tcx GlobalCtxt<'tcx>> {
-        self.gcx.compute(|| {
-            let krate = self.parse().steal();
-
-            passes::create_global_ctxt(
-                self.compiler,
-                krate,
-                &self.gcx_cell,
-                &self.arena,
-                &self.hir_arena,
-            )
-        })
     }
 }
 
@@ -192,16 +157,7 @@ impl Compiler {
     where
         F: for<'tcx> FnOnce(&'tcx Queries<'tcx>) -> T,
     {
-        // Must declare `_timer` first so that it is dropped after `queries`.
-        let _timer;
         let queries = Queries::new(self);
-        let ret = f(&queries);
-
-        // The timer's lifetime spans the dropping of `queries`, which contains
-        // the global context.
-        _timer = self.sess.timer("free_global_ctxt");
-        queries.finish();
-
-        ret
+        f(&queries)
     }
 }

--- a/compiler/rustc_middle/src/ty/context.rs
+++ b/compiler/rustc_middle/src/ty/context.rs
@@ -10,6 +10,7 @@ use std::cmp::Ordering;
 use std::hash::{Hash, Hasher};
 use std::marker::PhantomData;
 use std::ops::{Bound, Deref};
+use std::sync::OnceLock;
 use std::{fmt, iter, mem};
 
 use rustc_abi::{ExternAbi, FieldIdx, Layout, LayoutData, TargetDataLayout, VariantIdx};
@@ -1347,33 +1348,6 @@ pub struct GlobalCtxt<'tcx> {
 
     /// Stores memory for globals (statics/consts).
     pub(crate) alloc_map: Lock<interpret::AllocMap<'tcx>>,
-
-    current_gcx: CurrentGcx,
-}
-
-impl<'tcx> GlobalCtxt<'tcx> {
-    /// Installs `self` in a `TyCtxt` and `ImplicitCtxt` for the duration of
-    /// `f`.
-    pub fn enter<F, R>(&'tcx self, f: F) -> R
-    where
-        F: FnOnce(TyCtxt<'tcx>) -> R,
-    {
-        let icx = tls::ImplicitCtxt::new(self);
-
-        // Reset `current_gcx` to `None` when we exit.
-        let _on_drop = defer(move || {
-            *self.current_gcx.value.write() = None;
-        });
-
-        // Set this `GlobalCtxt` as the current one.
-        {
-            let mut guard = self.current_gcx.value.write();
-            assert!(guard.is_none(), "no `GlobalCtxt` is currently set");
-            *guard = Some(self as *const _ as *const ());
-        }
-
-        tls::enter_context(&icx, || f(icx.tcx))
-    }
 }
 
 /// This is used to get a reference to a `GlobalCtxt` if one is available.
@@ -1517,7 +1491,8 @@ impl<'tcx> TyCtxt<'tcx> {
     /// By only providing the `TyCtxt` inside of the closure we enforce that the type
     /// context and any interned value (types, args, etc.) can only be used while `ty::tls`
     /// has a valid reference to the context, to allow formatting values that need it.
-    pub fn create_global_ctxt(
+    pub fn create_global_ctxt<T>(
+        gcx_cell: &'tcx OnceLock<GlobalCtxt<'tcx>>,
         s: &'tcx Session,
         crate_types: Vec<CrateType>,
         stable_crate_id: StableCrateId,
@@ -1529,7 +1504,8 @@ impl<'tcx> TyCtxt<'tcx> {
         query_system: QuerySystem<'tcx>,
         hooks: crate::hooks::Providers,
         current_gcx: CurrentGcx,
-    ) -> GlobalCtxt<'tcx> {
+        f: impl FnOnce(TyCtxt<'tcx>) -> T,
+    ) -> T {
         let data_layout = s.target.parse_data_layout().unwrap_or_else(|err| {
             s.dcx().emit_fatal(err);
         });
@@ -1538,7 +1514,7 @@ impl<'tcx> TyCtxt<'tcx> {
         let common_lifetimes = CommonLifetimes::new(&interners);
         let common_consts = CommonConsts::new(&interners, &common_types, s, &untracked);
 
-        GlobalCtxt {
+        let gcx = gcx_cell.get_or_init(|| GlobalCtxt {
             sess: s,
             crate_types,
             stable_crate_id,
@@ -1562,8 +1538,23 @@ impl<'tcx> TyCtxt<'tcx> {
             canonical_param_env_cache: Default::default(),
             data_layout,
             alloc_map: Lock::new(interpret::AllocMap::new()),
-            current_gcx,
+        });
+
+        let icx = tls::ImplicitCtxt::new(&gcx);
+
+        // Reset `current_gcx` to `None` when we exit.
+        let _on_drop = defer(|| {
+            *current_gcx.value.write() = None;
+        });
+
+        // Set this `GlobalCtxt` as the current one.
+        {
+            let mut guard = current_gcx.value.write();
+            assert!(guard.is_none(), "no `GlobalCtxt` is currently set");
+            *guard = Some(&gcx as *const _ as *const ());
         }
+
+        tls::enter_context(&icx, || f(icx.tcx))
     }
 
     /// Obtain all lang items of this crate and all dependencies (recursively)

--- a/src/librustdoc/doctest.rs
+++ b/src/librustdoc/doctest.rs
@@ -175,23 +175,26 @@ pub(crate) fn run(dcx: DiagCtxtHandle<'_>, input: Input, options: RustdocOptions
         ..
     } = interface::run_compiler(config, |compiler| {
         compiler.enter(|queries| {
-            let collector = queries.global_ctxt().enter(|tcx| {
-                let crate_name = tcx.crate_name(LOCAL_CRATE).to_string();
-                let crate_attrs = tcx.hir().attrs(CRATE_HIR_ID);
-                let opts = scrape_test_config(crate_name, crate_attrs, args_path);
-                let enable_per_target_ignores = options.enable_per_target_ignores;
+            let krate = queries.parse().steal();
 
-                let mut collector = CreateRunnableDocTests::new(options, opts);
-                let hir_collector = HirCollector::new(
-                    ErrorCodes::from(compiler.sess.opts.unstable_features.is_nightly_build()),
-                    enable_per_target_ignores,
-                    tcx,
-                );
-                let tests = hir_collector.collect_crate();
-                tests.into_iter().for_each(|t| collector.add_test(t));
+            let collector =
+                rustc_interface::create_and_enter_global_ctxt(&compiler, krate, |tcx| {
+                    let crate_name = tcx.crate_name(LOCAL_CRATE).to_string();
+                    let crate_attrs = tcx.hir().attrs(CRATE_HIR_ID);
+                    let opts = scrape_test_config(crate_name, crate_attrs, args_path);
+                    let enable_per_target_ignores = options.enable_per_target_ignores;
 
-                collector
-            });
+                    let mut collector = CreateRunnableDocTests::new(options, opts);
+                    let hir_collector = HirCollector::new(
+                        ErrorCodes::from(compiler.sess.opts.unstable_features.is_nightly_build()),
+                        enable_per_target_ignores,
+                        tcx,
+                    );
+                    let tests = hir_collector.collect_crate();
+                    tests.into_iter().for_each(|t| collector.add_test(t));
+
+                    collector
+                });
             compiler.sess.dcx().abort_if_errors();
 
             collector

--- a/src/librustdoc/lib.rs
+++ b/src/librustdoc/lib.rs
@@ -857,12 +857,12 @@ fn main_args(
         }
 
         compiler.enter(|queries| {
-            let mut gcx = queries.global_ctxt();
+            let krate = queries.parse().steal();
             if sess.dcx().has_errors().is_some() {
                 sess.dcx().fatal("Compilation failed, aborting rustdoc");
             }
 
-            gcx.enter(|tcx| {
+            rustc_interface::create_and_enter_global_ctxt(&compiler, krate, |tcx| {
                 let (krate, render_opts, mut cache) = sess.time("run_global_ctxt", || {
                     core::run_global_ctxt(tcx, show_coverage, render_options, output_format)
                 });

